src/modm/platform/timer/stm32/general_purpose.cpp.in

```diff
@@ -172,3 +172,31 @@ modm::platform::Timer{{ id }}::enableInterruptVector(bool enable, uint32_t prior
 	}
 %% endfor
 }
+
+// ----------------------------------------------------------------------------
+%% if ((id <= 5) or (id in [8, 9, 12]))
+bool
+modm::platform::Timer{{ id }}::isChannelConfiguredAsInput(uint32_t channel)
+{
+	bool isInput = false;
+	switch (channel) {
+		case 1:
+			isInput = TIM{{ id }}->CCMR1 bitand TIM_CCMR1_CC1S;
+			break;
+		case 2:
+			isInput = TIM{{ id }}->CCMR1 bitand TIM_CCMR1_CC2S;
+			break;
+%% if (id <= 5)
+		case 3:
+			isInput = TIM{{ id }}->CCMR2 bitand TIM_CCMR2_CC3S;
```

Comment:
Please use `&` instead of `bitand`. The idea is to visually distinguish bit operations and logic operations by using `&`, '|`, etc for bit operations and `and`, `or`, etc for logic operations. `bitand` kinda messes this up.